tests/kernel/interrupt/CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
+project(NONE)
+
+if(NOT CONFIG_RISCV32)
```

Comment:
this is confusing.
just ifdef it out in irq_lock.c
why do we need separate C files anyway?

Reply:
irq_lock.c contains the test cases specific to validate irq_lock() api. irq_lock specific test case is failing for the RISV32 architecture, looks like irq_lock() is not behaving as expected. We are excluding irq_lock.c file only for RISCV32 architecture. 
Will create a git hub issue once this test application get merged.

Reply:
No, don't do this, consolidate these files and ifdef out the case if it doesn't work on some arch.